### PR TITLE
Creating a no exceptions build for Envoy Mobile.

### PR DESCRIPTION
**Pull request by alyssawilk**
The initial passes of this were over 100 files of workarounds to build

bazel build --config=remote --define=admin_html=disabled --define=admin_functionality=disabled --define envoy_exceptions=disabled --define envoy_mobile_listener=disabled  --define=envoy_yaml=disabled  //test/performance:test_binary_size --copt=-fno-unwind-tables --copt=-fno-exceptions

Now we're in the low 70 files, but there's still work todo (see TODO). Landing now as E-M already panics on bad config, and we want to see what the binary size impact is for our E-M usage.

I will continue working on https://github.com/envoyproxy/envoy-mobile/issues/176 to do proper error handling with the end goal of removing this macro.

Risk Level: low
Testing: n/a
Docs Changes:
Release Notes:
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

File: source/common/network/listen_socket_impl.cc [hunk 1]
```diff
@@ -24,18 +24,22 @@ Api::SysCallIntResult ListenSocketImpl::bind(Network::Address::InstanceConstShar
...
-                                          connection_info_provider_->localAddress()->asString(),
-                                          errorDetails(result.errno_)),
-                              result.errno_);
+    const std::string error =
+        fmt::format("cannot bind '{}': {}", connection_info_provider_->localAddress()->asString(),
+                    errorDetails(result.errno_));
+#ifdef ENVOY_DISABLE_EXCEPTIONS
+    PANIC(error);
+#else
+    throw SocketBindException(error, result.errno_);
```
Comment (RyanTheOptimist) [inline source/common/network/listen_socket_impl.cc:33]: This can't use throwExceptionOrPanic() because this exception type's constructor is special?

**Review by RyanTheOptimist — APPROVED** [review]
Wow, looks great. Mostly mechanical. Woo hoo

File: source/common/network/listen_socket_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/network/listen_socket_impl.cc:33]: yeah, I tried to normalize the constructor like I did elsewhere but it turns out we use the errno in tests to keep listening on random ports until we get one that works so it's non-trivial

**Comment by alyssawilk** [comment]
Also tagging a !google !EM reviewer :-)

**Review by zuercher — APPROVED** [review]
lgtm

**Pull request merged**